src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkConfig.java

```diff
@@ -218,4 +220,31 @@ private void validateClusteringOrder(@Nullable Schema schema) {
       }
     }
   }
+
+  private void validateOperationProperties(@Nullable Schema schema) {
+    if (Arrays.stream(Operation.values()).map(Enum::name).noneMatch(operation.toUpperCase()::equals)) {
+      throw new InvalidConfigPropertyException(String.format("'%s' is incorrect value for field 'Operation'. " +
+                                                               "This field should contain one of the next values: " +
+                                                               "INSERT, UPDATE or UPSERT.", operation), NAME_OPERATION);
+    }
+    if (!Operation.INSERT.equals(getOperation()) && getRelationTableKey() == null) {
```

Comment:
though it's logically equivalent now, it's safer for this check to be on equality with UPDATE or UPSERT, in case new types are added in the future.

Reply:
Done.